crates/client/rpc/src/lib.rs

```diff
@@ -1226,18 +1254,18 @@ where
             FieldElement::default()
         };
 
+        let starknet_block_hash = BlockHash(starknet_block.header().hash::<H>().into());
+
+        let state_diff = self.get_state_diff(&starknet_block_hash).map_err(|e| {
+            error!("Failed to get state diff. Starknet block hash: {starknet_block_hash}, error: {e}");
```

Comment:
Coz here you will get a state diff, just a false one

Reply:
Fixed